sound/soc/sof/trace.c

```diff
@@ -13,6 +13,55 @@
 #include "sof-priv.h"
 #include "ops.h"
 
+static ssize_t sof_dfsentry_trace_filter_write(struct file *file,
+		const char __user *from, size_t count, loff_t *ppos)
+{
+	struct snd_sof_dfsentry *dfse = file->private_data;
+	struct snd_sof_dev *sdev = dfse->sdev;
+	char *string;
+	int ret;
+
+	string = kmalloc(count, GFP_KERNEL);
+	if (!string) {
+		ret = -ENOMEM;
+		goto error;
+	}
+
+	ret = simple_write_to_buffer(string, count, ppos, from, count);
+	if (ret != count)
+		ret = ret >= 0 ? -EIO : ret;
```

Comment:
Im not sure this check is really needed. Can we simple check if ret < 0 and return the return value directly?